lib/block.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const stream = require('stream');
+const StringifierStream = require('./streams/stringifier-stream');
+const Fragment = require('./fragment');
+const FRAGMENT_EVENTS = ['start', 'response', 'end', 'error', 'timeout', 'fallback', 'warn'];
+
+module.exports.buildBlock = function buildBlock(request, originalContext) {
+    // NOTE: should there be a separation between context as something global
+    // vs. context as quilt/tailor-specific term?
+    let context = originalContext;
+
+    const {
+        maxAssetLinks,
+        asyncStream,
+        handleTag,
+        requestFragment,
+
+        // Should the two below be one???
+        pipeDefinition,
+        pipeAttributes,
+        pipeInstanceName,
+    } = context;
+
+    const resultStream = new StringifierStream((tag) => {
+        const { placeholder, name } = tag;
+
+        const fetchFragment = (combinedContext) => {
+            const fragment = new Fragment({
+                tag,
+                context: combinedContext,
+                index: combinedContext.index,
+                requestFragment,
+                pipeInstanceName,
+                pipeAttributes,
+                maxAssetLinks
+            });
+
+            FRAGMENT_EVENTS.forEach((eventName) => {
+                fragment.on(eventName, (...args) => {
+                    const prefixedName = 'fragment:' + eventName;
+                    resultStream.emit(prefixedName, request, fragment.attributes, ...args);
+                });
+            });
+
+            const isAsync = fragment.attributes.async;
```

Comment:
shall we keep the code similar to how it was in request-handler? 

https://github.com/zalando/tailor/blob/7d0fff28be95c811af064c7d3290e36badc6a419/lib/request-handler.js#L105

Reply:
Agreed! 

Wow, how can you comment with code snippets like this? oO

Reply:
hehe.. there is a small option at the left end where you can copy the preview link once you click on the line.